package/gluon-mesh-vpn-wireguard/luasrc/lib/gluon/upgrade/405-wireguard

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/lua
+local site = require 'gluon.site'
+local util = require 'gluon.util'
+local uci = require('simple-uci').cursor()
+
+uci:delete('wireguard', 'mesh_vpn', 'user')
+
+local add_groups
+
+local function generate_wg_iface_name(peer)
+  return "mesh_wg_" .. peer
+end
+
+local function add_peer(group, name, config)
+  uci:section('network', 'interface', generate_wg_iface_name(name) .. 'm', {
+    type = 'wireguard',
+    ifname = generate_wg_iface_name(name),
+    proto = 'gluon_mesh',
+  })
+
+  uci:section('network', 'interface', generate_wg_iface_name(name) .. 's', {
+    ip6addr = 'fe80::2/64',
+    ifname = generate_wg_iface_name(name),
+    proto = 'static',
+  })
+
+  uci:section('network', 'interface', generate_wg_iface_name(name) .. 'w', {
+    ifname = generate_wg_iface_name(name),
+    proto = 'gluon_wireguard',
+  })
+
+  uci:section('wireguard', 'peer', group .. '_peer_' .. name, {
+    enabled = true,
+    net = 'mesh_vpn',
+    ifname = generate_wg_iface_name(name),
+    group = group,
+    key = config.key,
+    remote = config.remote,
+    allowedIPs = '::/0',
```

Comment:
Same here (IPv4).

Reply:
ok. we need to route the client_prefix and node_prefix and all other prefixes that are going to be shared with the net using the prefixd concept. So the tightest mask unfortunately is definitely close to ::/0.